fvm/crypto.go

```diff
@@ -149,14 +152,14 @@ func verifySignatureFromRuntime(
 
 	publicKey, err := crypto.DecodePublicKey(sigAlgo, rawPublicKey)
 	if err != nil {
-		// TODO: improve error passing https://github.com/onflow/cadence/issues/202
-		return false, err
+		issue := &errors.InvalidInputError{Err: fmt.Errorf("cannot decode public key: %w", err)}
```

Comment:
Another pattern we use in a few other packages ([eg](https://github.com/onflow/flow-go/blob/f0cd379b526e6d0b0fc467967d47d3531df6be8b/state/errors.go#L17-L21) is defining constructors similar to `Errorf` which handle creating the inner error:
```suggestion
		issue := errors.NewInvalidInputErrorf("cannot decode public key: %w", err)
```

Reply:
Applied at b24e3d4

Reply:
thought it is still not in the best shape but I think that requires some changes on places that generate those errors.